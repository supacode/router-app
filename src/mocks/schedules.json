[
  {
    "arrivalDate": "2020-08-29",
    "end": "12:00",
    "start": "09:00"
  },
  {
    "arrivalDate": "2021-10-29",
    "end": "10:00",
    "start": "15:00"
  }
]
