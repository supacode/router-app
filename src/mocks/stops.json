[
  {
    "address": "13353 Berlin, Germany",
    "company": "insta",
    "openingHours": {
      "from": 900,
      "to": 1700
    },
    "schedule": {
      "start": "2020-08-29 09:00:00",
      "end": "2020-08-29 12:00:00"
    }
  },
  {
    "address": "13305 Berlin, Germany",
    "company": "BASF",
    "openingHours": {
      "from": 900,
      "to": 1700
    },
    "schedule": {
      "start": "2020-08-29 13:00:00",
      "end": "2020-08-29 15:00:00"
    }
  },
  {
    "address": "10439 Berlin, Germany",
    "company": "DropOff GmbH",
    "openingHours": {
      "from": 900,
      "to": 1700
    },
    "schedule": {
      "start": "2020-08-29 16:00:00",
      "end": "2020-08-29 19:00:00"
    }
  },
  {
    "address": "10999 Berlin, Germany",
    "company": "InstaFreight GmbH",
    "openingHours": {
      "from": 700,
      "to": 1800
    },
    "schedule": {
      "start": "2020-08-31 09:00:00",
      "end": "2020-08-31 12:00:00"
    }
  },
  {
    "address": "10117 Berlin, Germany",
    "company": "InstaFreight New GmbH",
    "openingHours": {
      "from": 800,
      "to": 1800
    },
    "schedule": {
      "start": "2020-08-31 15:30:00",
      "end": "2020-08-31 18:45:00"
    }
  }
]
