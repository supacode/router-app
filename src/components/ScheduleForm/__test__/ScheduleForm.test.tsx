import { fireEvent, render, screen } from '@testing-library/react';
import { ScheduleForm } from '../';
import stops from '../../../mocks/stops.json';
import schedules from '../../../mocks/schedules.json';

const mockFormProps = {
  formId: 'a',
  isFixed: false,
  stop: stops[0],
  schedule: schedules[0],
  isFlexible: true,
  isFirstStop: true,
  canEditForm: true,
  isSemiFlexible: true,
  onFormSubmit: jest.fn(),
};

beforeEach(() => {
  jest.clearAllMocks();
});

describe('ScheduleForm', () => {
  it('should match snapshot', () => {
    const { container } = render(<ScheduleForm {...mockFormProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should submit if form is valid', () => {
    render(<ScheduleForm {...mockFormProps} />);

    const form = screen.getByTestId(mockFormProps.formId);

    fireEvent.submit(form);

    expect(mockFormProps.onFormSubmit).toHaveBeenCalledTimes(1);
  });
});
