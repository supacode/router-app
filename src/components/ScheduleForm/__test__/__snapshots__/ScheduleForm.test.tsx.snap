// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleForm should match snapshot 1`] = `
<div>
  <div
    class="schedule-form__wrapper"
  >
    <form
      class="schedule-form"
      data-testid="a"
      id="a"
      schedule-form-id="a"
    >
      <h3
        class="schedule-form__title"
      >
        Arrival Date
      </h3>
      <div
        class="schedule-form__input-wrapper"
      >
        <div
          class="input arrival-date-input"
        >
          <div
            class="input__input-wrapper"
          >
            <input
              class="input__input"
              id="a-arrival-date"
              name="arrivalDate"
              required=""
              type="date"
              value="2020-08-29"
            />
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="input__input-wrapper"
          >
            <label
              class="input__prepend"
              for="a-from"
            >
              Von
            </label>
            <input
              class="input__input"
              id="a-from"
              name="start"
              required=""
              type="time"
              value="09:00"
            />
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="input__input-wrapper"
          >
            <label
              class="input__prepend"
              for="a-to"
            >
              Bis
            </label>
            <input
              class="input__input"
              id="a-to"
              name="end"
              required=""
              type="time"
              value="12:00"
            />
          </div>
        </div>
        <div
          class="input arrival-date-input"
        >
          <div
            class="input__input-wrapper"
          >
            <input
              class="input__input"
              id="a-arrival-date"
              name="arrivalDate"
              required=""
              type="date"
              value="2020-08-29"
            />
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="input__input-wrapper"
          >
            <label
              class="input__prepend"
              for="a-from"
            >
              Von
            </label>
            <input
              class="input__input"
              id="a-from"
              name="start"
              required=""
              type="time"
              value="09:00"
            />
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="input__input-wrapper"
          >
            <label
              class="input__prepend"
              for="a-to"
            >
              Bis
            </label>
            <input
              class="input__input"
              id="a-to"
              name="end"
              required=""
              type="time"
              value="12:00"
            />
          </div>
        </div>
      </div>
      <div
        class="schedule-form__control"
      >
        <button
          class="button button__tertiary"
          type="button"
        >
          Cancel
        </button>
        <button
          class="button button__primary"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
